frontend/lib/wallets/WalletConnect.ts

```diff
@@ -1,32 +1,97 @@
+import WalletConnectProvider from '@walletconnect/web3-provider';
+import { ethers } from 'ethers';
+import { setupRpcUrlAndGetNetworks } from 'auctions-core/src/rpc';
+import { formatToHexWithoutPad } from 'auctions-core/helpers/format';
+import { getChainIdByNetworkType, getNetworkTypeByChainId } from 'auctions-core/src/network';
+import { setSigner } from 'auctions-core/src/signer';
 import WalletConnectLogo from '~/assets/icons/wallets/walletconnect.svg';
 import AbstractWallet from '~/lib/wallets/AbstractWallet';
 
 export default class WalletConnect extends AbstractWallet {
     public static title = 'WalletConnect';
     public static icon = WalletConnectLogo;
-    public static downloadUrl = 'https://metamask.io/';
 
-    get isInterfaceReady() {
-        return true;
+    private static provider?: WalletConnectProvider;
+
+    private addresses?: string[];
+
+    public get address() {
+        if (Array.isArray(this.addresses) && this.addresses[0]) {
+            return this.addresses[0].toLowerCase();
+        }
+    }
+
+    async getProvider(): Promise<ethers.providers.JsonRpcProvider> {
+        if (!WalletConnect.provider) {
+            const rpcUrl = process.env.RPC_URL;
+            const { networks } = await setupRpcUrlAndGetNetworks(rpcUrl);
```

Comment:
If you only need networks, you can just call `getNetworks` here, since setup will also do other things that you don't need.
```suggestion
            const networks = awat getNetworks();
```